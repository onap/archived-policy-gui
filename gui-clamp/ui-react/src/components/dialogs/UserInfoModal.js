/*-
 * ============LICENSE_START=======================================================
 * ONAP CLAMP
 * ================================================================================
 * Copyright (C) 2019 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */

import React from 'react'
import Button from 'react-bootstrap/Button';
import Modal from 'react-bootstrap/Modal';
import Form from 'react-bootstrap/Form';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import styled from 'styled-components';
import UserService from '../../api/UserService';

const ModalStyled = styled(Modal)`
	background-color: transparent;
`

export default class UserInfoModal extends React.Component {

	constructor(props, context) {
		super(props, context);

		this.handleClose = this.handleClose.bind(this);
	    this.renderPermissions = this.renderPermissions.bind(this);
		this.renderUserName = this.renderUserName.bind(this);
		this.state = {
			show: true,
			userInfo: {}
		};
	}
	componentWillMount() {
		UserService.getUserInfo().then(userInfo => {
			this.setState({ userInfo: userInfo })
		});
	}

	handleClose() {
			this.props.history.push('/');
	}
	renderPermissions() {
          if (this.state.userInfo["allPermissions"]) {
	        var listOfPermissions = this.state.userInfo["allPermissions"].map(function(perm) {
	            return <Form.Control plaintext readOnly defaultValue={perm} />;
	        })
		    return listOfPermissions;
		  } else {
		    return;
		  }
 	}
	renderUserName() {
		if (this.state.userInfo["userName"]) {
			return <Form.Control plaintext readOnly defaultValue={this.state.userInfo["userName"]} />
		} else  {
			return;
		}
	}
	renderVersion() {
		if (this.state.userInfo["cldsVersion"]) {
			return <Form.Control plaintext readOnly defaultValue={this.state.userInfo["cldsVersion"]} />
		} else  {
			return;
		}
	}
	render() {
		return (
			<ModalStyled size="lg"  show={this.state.show} onHide={this.handleClose}>
				<Modal.Header closeButton>
					<Modal.Title>User Info</Modal.Title>
				</Modal.Header>
				<Modal.Body>
					<Form.Group as={Row} controlId="userName">
						<Form.Label column sm="3">Current User:</Form.Label>
						<Col>{this.renderUserName()}</Col>
					</Form.Group>
					<Form.Group as={Row} controlId="cldsVersion">
						<Form.Label column sm="3">CLDS Version:</Form.Label>
						<Col>{this.renderVersion()}</Col>
					</Form.Group>
					<Form.Group as={Row} controlId="userPermissions">
						<Form.Label column sm="3">User Permissions:</Form.Label>
						<Col>
							{this.renderPermissions()}
						</Col>
					</Form.Group>
				</Modal.Body>
				<Modal.Footer>
					<Button variant="secondary" type="null" onClick={this.handleClose}>Cancel</Button>
				</Modal.Footer>
			</ModalStyled>
		);
	}
}
