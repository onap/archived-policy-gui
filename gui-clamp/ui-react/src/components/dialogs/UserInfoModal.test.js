/*-
 * ============LICENSE_START=======================================================
 * ONAP CLAMP
 * ================================================================================
 * Copyright (C) 2019 AT&T Intellectual Property. All rights
 *                             reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */
import React from 'react';
import { shallow } from 'enzyme';
import UserInfoModal from './UserInfoModal';

describe('Verify UserInfoModal', () => {

	beforeEach(() => {
		fetch.resetMocks();
		fetch.mockImplementation(() => {
			return Promise.resolve({
				ok: true,
				status: 200,
				json: () => {
					return Promise.resolve({
						"userName": "test",
						"cldsVersion": "1.0.0"
					});
			}});
		});
	})

	it('Test the render method full permission', () => {
		const component = shallow(<UserInfoModal />)
		component.setState({ userInfo: {
			"userName": "test",
			"cldsVersion": "1.0.0",
			"allPermissions": ["permission1","permission2"]
		}});
		expect(component).toMatchSnapshot();
	});

	it('Test the render method no permission', () => {
		const component = shallow(<UserInfoModal />)
		component.setState({ userInfo: {}
		});

		expect(component.find('FormControl').length).toEqual(0);
	});

	it('Test the render method read permission', () => {
		const component = shallow(<UserInfoModal />)
		component.setState({ userInfo: {
			"userName": "test",
			"cldsVersion": "1.0.0",
			"allPermissions": ["permission1","permission2"]
		}});

		expect(component.find('FormControl').length).toEqual(4);

		const forms = component.find('FormControl');
		expect(forms.get(0).props.defaultValue).toEqual("test");
		expect(forms.get(1).props.defaultValue).toEqual("1.0.0");
		expect(forms.get(2).props.defaultValue).toEqual("permission1");
		expect(forms.get(3).props.defaultValue).toEqual("permission2");
	});
});
