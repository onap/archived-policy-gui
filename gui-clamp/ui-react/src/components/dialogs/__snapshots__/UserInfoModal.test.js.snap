// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify UserInfoModal Test the render method full permission 1`] = `
<Styled(Modal)
  onHide={[Function]}
  show={true}
  size="lg"
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      User Info
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <FormGroup
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "noGutters": false,
          },
          "render": [Function],
        }
      }
      controlId="userName"
    >
      <FormLabel
        column={true}
        sm="3"
        srOnly={false}
      >
        Current User:
      </FormLabel>
      <Col>
        <FormControl
          defaultValue="test"
          plaintext={true}
          readOnly={true}
        />
      </Col>
    </FormGroup>
    <FormGroup
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "noGutters": false,
          },
          "render": [Function],
        }
      }
      controlId="cldsVersion"
    >
      <FormLabel
        column={true}
        sm="3"
        srOnly={false}
      >
        CLDS Version:
      </FormLabel>
      <Col>
        <FormControl
          defaultValue="1.0.0"
          plaintext={true}
          readOnly={true}
        />
      </Col>
    </FormGroup>
    <FormGroup
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "noGutters": false,
          },
          "render": [Function],
        }
      }
      controlId="userPermissions"
    >
      <FormLabel
        column={true}
        sm="3"
        srOnly={false}
      >
        User Permissions:
      </FormLabel>
      <Col>
        <FormControl
          defaultValue="permission1"
          key="permission1"
          plaintext={true}
          readOnly={true}
        />
        <FormControl
          defaultValue="permission2"
          key="permission2"
          plaintext={true}
          readOnly={true}
        />
      </Col>
    </FormGroup>
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="null"
      variant="secondary"
    >
      Cancel
    </Button>
  </ModalFooter>
</Styled(Modal)>
`;
